### ISSUE TITLE
Error when rebuilding with electron-rebuild.cmd

### ISSUE THREAD

**Issue by MY020537** [tojocky/node-printer#282]
Recently i upgraded electron application with Node version 12.13.0 and Electron 8.0.0. Printer module started causing issues once again :(  
Found out to run electron-rebuild.cmd. but getting list of issues like 

```
**node_printer_win.cc(622): error C2661: 'v8::Object::Set': no overloaded function takes 2 arguments**
```

Is it because of Node version ? How can i fix this ? Kindly help

**Comment by JanDevDE** [comment]
Same problem.
I'm running the following dependencies:

`"dependencies": {
    "printer": "^0.4.0",
  },
  "devDependencies": {
    "electron": "^8.0.1",
    "electron-rebuild": "^1.10.0"
  }`

After using --build-from-source I get the following error:

`\node_modules\printer\lib\node_printer.node'
was compiled against a different Node.js version using
NODE_MODULE_VERSION 72. This version of Node.js requires
NODE_MODULE_VERSION 76. Please try re-compiling or re-installing
the module (for instance, using `npm rebuild` or `npm install`).`

I tried to resolve this with using electron-rebuild, but sadly I get compile errors as described from @MY020537

**Comment by JanDevDE** [comment]
Just FYI, when going down some major electron versions, namely to version 6.1.7 and then running node-gyp.cmd rebuild --target=6.1.7 --arch=x64 --dist-url=https://electronjs.org/headers in node_modules/printer you can get it running though. @MY020537

**Comment by MY020537** [comment]
@JanDevDE  Yeah i thought of downgrading But with that version of electron, we have got known bugs with ```webContents.print()``` api implementation. and i need that too considering functionality implementation. so can not choose that version either.

**Comment by MY020537** [comment]
@JanDevDE  - Which node version you are using with electron 6.1.7 ?

**Comment by xsenechal** [comment]
Same problem.
I came to the same conclusion and downgrade with success to electron v6.1.7, which uses NodeJS v12.4.0 (process.version from inside electron).  If I understand correctly, electron comes with its own NodeJS version, this the need to launch electron-rebuild.

**Comment by JanDevDE** [comment]
@MY020537 Ah, okay. I'm using v12.16.1, but that doesn't really matter IIRC.

**Comment by xuehaoze** [comment]
same issue here. 

Probably, we can change the "msvs_version": to 2013?

**Comment by xuehaoze** [comment]
tried vs 2013, seems doesn't work. 

And confirmed that the error gone after I downgraded the Electron version from v7.1.12 to v6.1.7.

**Comment by angjelonuho** [comment]
I have thwe same problem and only downgrading worked for me. I would appreciate fixing this issue on an upcoming update.
